app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -182,7 +182,7 @@ def on_error(error)
       end
       errored_element = @errored_element if defined?(@errored_element)
     ensure
-      MaintenanceTasks.error_handler.call(error, task_context, errored_element)
+      MaintenanceTasks.error_handler.call(error, task_context || {}, errored_element)
```

Comment:
What do you think about assigning `task_context = {}` at the start of this method instead? That would let us drop the else branch above as well.

```ruby
def on_error(error)
  task_context = {}

  @ticker.persist if defined?(@ticker)

  if defined?(@run)
    ...
    task_context = { ... }
  end
  ...
ensure
  MaintenanceTasks.error_handler.call(error, task_context, errored_element)
end

Reply:
Good idea, that makes it much cleaner 👍